File: app/src/main/java/com/concrete/android/challenge/ui/base/BaseViewHolder.java
package com.concrete.android.challenge.ui.base;

import android.view.View;
import androidx.recyclerview.widget.RecyclerView;

/**
 * @author Thiago Corredo
 * @since 2019-05-28
 */
public abstract class BaseViewHolder extends RecyclerView.ViewHolder {

  public BaseViewHolder(View itemView) {
    super(itemView);
  }

  public abstract void onBind(int position);
}

File: app/src/main/java/com/pedrenrique/githubapp/core/platform/BaseViewHolder.kt
package com.pedrenrique.githubapp.core.platform

import android.view.View
import androidx.recyclerview.widget.RecyclerView

abstract class BaseViewHolder<T>(view: View) : RecyclerView.ViewHolder(view) {
    abstract fun bind(item: T, typesFactory: TypesFactory)
}